src/backend/distributed/commands/utility_hook.c

```diff
@@ -279,6 +279,19 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 			ddlJobs = PlanIndexStmt((IndexStmt *) parsetree, queryString);
 		}
 
+		if (IsA(parsetree, ReindexStmt))
+		{
+			MemoryContext oldContext = MemoryContextSwitchTo(GetMemoryChunkContext(
```

Comment:
1) Why do we copy the parse tree? 

2) Also, it is a little bit confusing that `PlanReindexStmt ` returns a list. Maybe re-name it to `ErrorIfReindexOnDistributedTable()` etc and return void?